include/mbedtls/threading.h

```diff
@@ -47,6 +47,19 @@ typedef struct mbedtls_threading_mutex_t {
 } mbedtls_threading_mutex_t;
 #endif
 
+#if defined(MBEDTLS_THREADING_SRWLOCK)
+/*
+ * Use spaces to pass the macroses name check.
+ */
+#   define WIN32_LEAN_AND_MEAN
```

Comment:
Please add a comment explaining why `WIN32_LEAN_AND_MEAN` is good to have. After 30 seconds on the web, I guess it's [to avoid conflicts with some other headers](https://stackoverflow.com/questions/11040133/what-does-defining-win32-lean-and-mean-exclude-exactly#comment108482188_11040230)?

(For `_WIN32_WINNT`, even without knowing anything about Windows programming, it's obvious to me that it specifies the API version that we're requesting, similar to `#define _POSIX_C_SOURCE` on Unix-like systems.)